src/utils/components/ApplyStorageProfileSettings/utils.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import { TFunction } from 'react-i18next';
+import { uniq } from 'lodash';
+
+import {
+  V1beta1StorageSpecAccessModesEnum,
+  V1beta1StorageSpecVolumeModeEnum,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { t } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { ClaimPropertySets } from '@kubevirt-utils/types/storage';
+
+export const tooltipForOtherRecommended = (priority: number, t: TFunction): string => {
+  switch (priority) {
+    case 1:
+      return t('The 2nd most suggested option in StorageProfile.');
+    case 2:
+      return t('The 3rd most suggested option in StorageProfile.');
+    default:
+      return t('The {{priority}}th most suggested option in StorageProfile.', {
+        priority: priority + 1,
+      });
+  }
+};
+
+export const fromPriorityToLabels = (
+  priority: number,
+  recommendationCount: number,
+  t: TFunction,
```

Comment:
no need for t, the console context of trans will trigger all plugins rerender, in this loading, t will be recalculated (what we talk about)

Reply:
@metalice I'll test that behavior and post the results here.

Reply:
@metalice  it seems we have a bug here. Tested with https://github.com/kubevirt-ui/kubevirt-plugin/pull/2665 and as a result opened https://issues.redhat.com/browse/CNV-61822